src/main/java/com/raoulvdberge/refinedstorage/apiimpl/network/node/NetworkNodeCrafter.java

```diff
@@ -222,15 +253,48 @@ public boolean hasConnectivityState() {
         return true;
     }
 
+    @Override
+    @Nullable
+    public ICraftingPatternContainer getProxyPatternContainer() {
+        if (visited) {
+            return null;
+        }
+
+        INetworkNode facing = API.instance().getNetworkNodeManager(world).getNode(pos.offset(getDirection()));
+        if (!(facing instanceof ICraftingPatternContainer) || facing.getNetwork() != network) {
+            return this;
+        }
+
+        visited = true;
+        ICraftingPatternContainer facingContainer = ((ICraftingPatternContainer)facing).getProxyPatternContainer();
+        visited = false;
+        return facingContainer;
+    }
+
     @Override
     public boolean isBlocked() {
-        return blocked;
+        ICraftingPatternContainer proxy = getProxyPatternContainer();
+        return proxy != null && network != null && network.getCraftingManager().isContainerBlocked(proxy.getUuid());
     }
 
     @Override
-    public void setBlocked(boolean blocked) {
-        this.blocked = blocked;
+    public void setBlocked(INetwork network, boolean blocked) {
```

Comment:
With the extra `network` parameter, and this method having no side-effects for `NetworkNodeCrafter` anymore, I feel like this method can be removed and moved to `CraftingManager`.